app/src/main/java/com/infomaniak/drive/data/services/BulkDownloadWorker.kt

```diff
@@ -0,0 +1,286 @@
+                    Result.failure()
+                }
+                is RemoteFileException -> {
+                    Sentry.captureException(exception)
+                    Result.failure()
+                }
+                else -> {
+                    SentryLog.e(TAG, "Failure", exception)
+                    Result.failure()
+                }
+            }
+        }.also {
+            notificationManagerCompat.cancel(NOTIFICATION_ID)
+        }
+    }
+
+    private fun clearFiles() {
+        for ((_, value) in filesPair.entries) {
+            Pair(value.first, value.second).letAll { file, offlineFile ->
+                if (offlineFile.exists() && !file.isIntactFile(offlineFile)) offlineFile.delete()
+            }
+        }
+    }
+
+    private suspend fun initOfflineDownload(): Result {
+        var result = Result.failure()
+        fileIds.forEach { fileId ->
+            val file = FileController.getFileById(fileId, userDrive)
+            val offlineFile = file?.getOfflineFile(applicationContext, userDrive.userId)
+            val cacheFile = file?.getCacheFile(applicationContext, userDrive)
+
+            offlineFile?.let {
+                if (file.isOfflineAndIntact(it)) {
+                    result = Result.success()
+                    return@forEach
+                }
+            }
+
+            setProgress(workDataOf(PROGRESS to 0, FILE_ID to fileId))
+
+            if (offlineFile?.exists() == true) offlineFile.delete()
+            if (cacheFile?.exists() == true) cacheFile.delete()
+
+            if (file == null || offlineFile == null) {
+                getFileFromRemote(fileId, userDrive) { downloadedFile ->
+                    filesPair[downloadedFile.id] =
+                        Pair(downloadedFile, downloadedFile.getOfflineFile(applicationContext, userDrive.driveId))
+                }
+            } else {
+                filesPair[file.id] = Pair(file, offlineFile)
+            }
+
+            result = file?.let {
+                startOfflineDownload(file = file, offlineFile = offlineFile!!)
+            } ?: Result.failure()
+        }
+
+        if (result == Result.success()) {
+            applicationContext.cancelNotification(NOTIFICATION_ID)
+        }
+
+        return result
+    }
+
+    private fun createDownloadNotification(): NotificationCompat.Builder {
+        val cancelPendingIntent = WorkManager.getInstance(applicationContext).createCancelPendingIntent(id)
+        val cancelAction = NotificationCompat.Action(
+            /* icon = */ null,
+            /* title = */ applicationContext.getString(R.string.buttonCancel),
+            /* intent = */ cancelPendingIntent
+        )
+        return applicationContext.downloadProgressNotification().apply {
+            setOngoing(true)
+            setContentTitle(applicationContext.getString(R.string.bulkDownloadNotificationTitleNoProgress))
+            addAction(cancelAction)
+        }
+    }
```

Comment:
This function already exists in `DownloadWorker`, please avoid duplication

Reply:
I agree with you, I'll merge what's existing in `DownloadWorker` and `BulkDownloadWorker`.